### PR TITLE
redid the task

### PR DESCRIPTION
**Pull request by MaksOnofriychuk**

### CODE REVIEW

File: src/components/Table/AvarageSumOfRow.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import { useTypedSelector } from "../../hooks/useTypedSelector";
+import { MatrixCell, MatrixCells } from "../../types/table";
+
+const AvarageSumOfRow = () => {
+  const { matrix } = useTypedSelector((state) => state.table);
+
+  const rowSums = matrix.map((columns: MatrixCells) => {
```
Comment (apiloian) [inline src/components/Table/AvarageSumOfRow.tsx]: rowSum: number[]

File: src/components/Form/Form.tsx [hunk 2]
```diff
@@ -1,19 +1,24 @@
...
+  columns: string;
+  rows: string;
+  cells: string;
+}
+
+const Form: FC = () => {
   const disptach = useDispatch();
 
-  const [formState, setFormState] = useState({
+  const [formState, setFormState] = useState<FormState>({
```
Comment (apiloian) [inline src/components/Form/Form.tsx:14]: redux

File: src/components/Form/Form.tsx [hunk 3]
```diff
@@ -30,7 +35,8 @@ const Form = () => {
     });
   };
 
-  const handleChange = ({ target }) => {
+  const handleChange = (event: React.FormEvent<HTMLInputElement>) => {
+    const target = event.target as HTMLInputElement;
```
Comment (apiloian) [inline src/components/Form/Form.tsx:38]: ts

File: src/components/Table/AvarageSumOfRow.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    return columns.value.reduce(
+      (resultNumber: number, cell: MatrixCell) => resultNumber + cell.value,
+      0
+    );
+  });
+
+  const sumColumnsSum = rowSums.reduce(
```
Comment (apiloian) [inline src/components/Table/AvarageSumOfRow.tsx]: add new

File: src/components/Table/Columns.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  CellProps,
+  MatrixCell,
+  MatrixCells,
+  RowAvarageProps,
+} from "../../types/table";
+
+export const Cell: FC<CellProps> = ({ id, value }) => {
+  const dispatch = useDispatch();
+
+  const addingCell = (id: any) => {
```
Comment (apiloian) [inline src/components/Table/Columns.tsx]: func

File: src/components/Form/Form.tsx (hunk 3 above)
Comment (MaksOnofriychuk) [reply src/components/Form/Form.tsx:38]: swear without it

File: src/components/Table/AvarageSumOfRow.tsx (hunk 1 above)
Comment (MaksOnofriychuk) [reply src/components/Table/AvarageSumOfRow.tsx]: +

**Review by arturiop — APPROVED** [review]

**Pull request merged**